tensorflow/core/kernels/one_hot_op.cc

```diff
@@ -113,45 +113,77 @@ class OneHotOp : public OpKernel {
   TF_DISALLOW_COPY_AND_ASSIGN(OneHotOp);
 };
 
-#define REGISTER_ONE_HOT(type)                           \
+#define REGISTER_ONE_HOT_32(type)                        \
```

Comment:
you can always just have both int32 and int64 definitions inside a single "#define".

you can also "#define REGISTER_ONE_HOT_INDEX(type, index_type) ...." then "#define REGISTER_ONE_HOT(type) REGISTER_ONE_HOT_INDEX(type, int32); REGISTER_ONE_HOT_INDEX(type, int64)